### ISSUE TITLE
'Import CSV' does not trigger event hooks

### ISSUE THREAD

**Issue by arladmin** [directus/directus#20284]
### Describe the Bug

I have a flow whose trigger is configured like so:

![image](https://github.com/directus/directus/assets/64059154/83a75dbb-1798-48d5-a1a4-3ac6b7129c68)

But, whenever i import data from a csv file into the concerned collection, the flow doesn't get triggered at all!

### To Reproduce

^

### Directus Version

v10.7.1

### Hosting Strategy

Self-Hosted (Docker Image)

**Comment by azrikahar** [comment]
Seems like this indeed aren't working anymore after #19691 which moved the import workload to worker threads.

Doing a partial revert of said PR like the following as a quick test, seems to confirm that the flow will once again get triggered:

```diff
diff --git a/api/src/controllers/utils.ts b/api/src/controllers/utils.ts
index c0137505e4..2a74839916 100644
--- a/api/src/controllers/utils.ts
+++ b/api/src/controllers/utils.ts
@@ -8,7 +8,7 @@ import { InvalidPayloadError, InvalidQueryError, UnsupportedMediaTypeError } fro
 import collectionExists from '../middleware/collection-exists.js';
 import { respond } from '../middleware/respond.js';
 import type { ImportWorkerData } from '../services/import-export/import-worker.js';
-import { ExportService } from '../services/import-export/index.js';
+import { ExportService, ImportService } from '../services/import-export/index.js';
 import { RevisionsService } from '../services/revisions.js';
 import { UtilsService } from '../services/utils.js';
 import asyncHandler from '../utils/async-handler.js';
@@ -107,6 +107,11 @@ router.post(
                        throw new UnsupportedMediaTypeError({ mediaType: req.headers['content-type']!, where: 'Content-Type header' });
                }

+               const service = new ImportService({
+                       accountability: req.accountability,
+                       schema: req.schema,
+               });
+
                let headers;

                if (req.headers['content-type']) {
@@ -121,38 +126,13 @@ router.post(
                const busboy = Busboy({ headers });

                busboy.on('file', async (_fieldname, fileStream, { mimeType }) => {
-                       const { createTmpFile } = await import('@directus/utils/node');
-                       const { getWorkerPool } = await import('../worker-pool.js');
-
-                       const tmpFile = await createTmpFile().catch(() => null);
-
-                       if (!tmpFile) throw new Error('Failed to create temporary file for import');
-
-                       fileStream.pipe(fs.createWriteStream(tmpFile.path));
-
-                       fileStream.on('end', async () => {
-                               const workerPool = getWorkerPool();
-
-                               const require = createRequire(import.meta.url);
-                               const filename = require.resolve('../services/import-export/import-worker');
-
-                               const workerData: ImportWorkerData = {
-                                       collection: req.params['collection']!,
-                                       mimeType,
-                                       filePath: tmpFile.path,
-                                       accountability: req.accountability,
-                                       schema: req.schema,
-                               };
-
-                               try {
-                                       await workerPool.run(workerData, { filename });
-                                       res.status(200).end();
-                               } catch (error) {
-                                       next(error);
-                               } finally {
-                                       await tmpFile.cleanup();
-                               }
-                       });
+                       try {
+                               await service.import(req.params['collection']!, mimeType, fileStream);
+                       } catch (err: any) {
+                               return next(err);
+                       }
+
+                       return res.status(200).end();
                });

                busboy.on('error', (err: Error) => next(err));
```

**Comment by arladmin** [comment]
Has this issue not been resolved in the v10.7.2 release?

**Comment by hermanndettmann** [comment]
@arladmin unfortunately not.

**Comment by azrikahar** [comment]
> Has this issue not been resolved in the v10.7.2 release?

It is still open as it has yet to be worked on.

> That's strange.
Normally, this type of a breaking issue would be fixed quickly, with a hot fix, rather than not even being resolved in a subsequent release.

Understandably this is important to you and all affected users 👍 I haven't started looking into this issue myself, but I'll try to check it out after resolving the ones that I'm assigned to, or if someone else has the time to tackle this issue before that.

**Comment by asperling** [comment]
From what I read the bug is considered to be breaking. Then there is a PR ready since 8 weeks ago. So might  I ask what the holdup is?
